### PR TITLE
SciML Organization Introduction post

### PR DESCRIPTION
**Pull request by ChrisRackauckas**

### CODE REVIEW

File: _posts/2020-3-27-SciML.md [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+to find more behaviors and handle those as well. Here is a current benchmark
+showcasing native DifferentialEquations.jl methods outperforming classical Fortran
+methods like LSODA by 5x on a 20 equation stiff ODE benchmark:
+
+![](https://user-images.githubusercontent.com/1814174/77687352-a0082800-6f74-11ea-924d-442a0836be6d.PNG)
+[Reference: Pollution Model Benchmarks](https://benchmarks.sciml.ai/html/StiffODE/Pollution.html)
+
+### We provide tools for deriving and fitting scientific models
+
+It is very rare that someone things their model is perfect. Instead, a large portion
```
Comment (ViralBShah) [inline _posts/2020-3-27-SciML.md]: thinks

Comment (ViralBShah) [reply _posts/2020-3-27-SciML.md]: Instead doesn't seem to flow well

File: _posts/2020-3-27-SciML.md [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+of the focus of our organization is to help scientific modelers derive equations
+and fit models. This includes tools for:
+
+- [Optimization-based and Bayesian parameter estimation](https://docs.SciML.org/dev/analysis/parameter_estimation/)
```
Comment (ViralBShah) [inline _posts/2020-3-27-SciML.md]: Optimizaton-based something? Sounds incomplete

File: _posts/2020-3-27-SciML.md [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+differential equation solvers to scientific simulators and tools for automatically
+discovering the scientific models. What I wanted to do with this post was introduce
+the organization by explaining a few things:
+
+- What SciML provides
+- What our goals are
+- Our next steps
+- How you can join in the process
+
+## The Service that SciML Provides
```
Comment (ViralBShah) [inline _posts/2020-3-27-SciML.md]: Service -> Software?

File: _posts/2020-3-27-SciML.md [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+
+n = 600000
+lb = -ones(4)*π
+ub = ones(4)*π
+sampler = SobolSample()
+A,B = QuasiMonteCarlo.generate_design_matrices(n,lb,ub,sampler)
+res1 = gsa(ishi,Sobol(),A,B)
+```
+
+Our reorganization will make it more accessible for users to find our GSA methods
```
Comment (ViralBShah) [inline _posts/2020-3-27-SciML.md]: What is GSA?

File: _posts/2020-3-27-SciML.md [hunk 5]
```diff
@@ -0,0 +1,374 @@
...
+- [RigidBodySim.jl](https://github.com/JuliaRobotics/RigidBodySim.jl) for robotics simulations,
+- [QuantumOptics.jl](https://qojulia.org/)
+- [DynamicalSystems.jl](https://juliadynamics.github.io/DynamicalSystems.jl/latest/)
+
+We make it easy to generate models of multi-scale systems using tools like
+[MultiScaleArrays.jl](https://github.com/SciML/MultiScaleArrays.jl)
+
+![](https://user-images.githubusercontent.com/1814174/27211626-79fe1b9a-520f-11e7-87f1-1cb33da91609.PNG)
+
+and build compilers like [ModelingToolkit.jl](https://github.com/SciML/ModelingToolkit.jl)
```
Comment (ViralBShah) [inline _posts/2020-3-27-SciML.md]: Discuss some underlying optimizations in MTK here?

File: _posts/2020-3-27-SciML.md [hunk 6]
```diff
@@ -0,0 +1,374 @@
...
+maintained package implementations of these methodologies to either start using
+for your scientific research, or utilize as a starting point for furthering the
+methods of scientific machine learning. For this reason, the goal of the SciML
+OSS organization is to be a hub for the development of robust cross-language
+scientific machine learning software. **In order to make this a reality, we as
+an organization have to commit to the following principles**:
+
+### Everything that we build is compatible with automatic differentiation
+
+Putting an arbitrary piece of code from the SciML group into ae training loop of
```
Comment (ViralBShah) [inline _posts/2020-3-27-SciML.md]: ae

**Comment by MasonProtter** [comment]
Reads really great, I really like it!

File: _posts/2020-3-27-SciML.md [hunk 7]
```diff
@@ -0,0 +1,380 @@
...
+When you read a paper that is
+[mixing neural networks with differential equations](https://arxiv.org/abs/2001.04385)
+or [designing new neural networks that satisfy incompressibility for modeling Navier-Stokes](https://arxiv.org/abs/1903.00033),
+you should be able to go online and find tweakable, high quality, and highly
+maintained package implementations of these methodologies to either start using
+for your scientific research, or utilize as a starting point for furthering the
+methods of scientific machine learning. For this reason, the goal of the SciML
+OSS organization is to be a hub for the development of robust cross-language
+scientific machine learning software. **In order to make this a reality, we as
+an organization have to commit to the following principles**:
```
Comment (c42f) [inline _posts/2020-3-27-SciML.md]:
```suggestion
an organization commit to the following principles**:
```

Is this a statement of fact for libraries which can be considered part of SciML, or a call to action?

File: _posts/2020-3-27-SciML.md [hunk 8]
```diff
@@ -0,0 +1,380 @@
+---
+layout: post
+title:  "SciML: An Open Source Software Organization for Scientific Machine Learning"
+date:   2020-3-27 10:00:00
+categories:
+---
+
+Scientific Machine Learning, abbreviated SciML, has been taking the academic
+world by storm as a interesting blend of traditional scientific mechanistic
```
Comment (nilshg) [inline _posts/2020-3-27-SciML.md]:
```suggestion
world by storm as an interesting blend of traditional scientific mechanistic

```

File: _posts/2020-3-27-SciML.md [hunk 9]
```diff
@@ -0,0 +1,380 @@
...
+scientific machine learning software. **In order to make this a reality, we as
+an organization have to commit to the following principles**:
+
+### Everything that we build is compatible with automatic differentiation
+
+Putting an arbitrary piece of code from the SciML group into a training loop of
+some machine learning library like [Flux](https://fluxml.ai/) will naturally work.
+This means we plan to enforce coding styles that are compatible with language-wide differentiable programming
+tools like [Zygote](https://github.com/FluxML/Zygote.jl), or provide pre-defined
+forward/adjoint rules via derivative rule package [ChainRules.jl](https://github.com/JuliaDiff/ChainRules.jl).
```
Comment (nilshg) [inline _posts/2020-3-27-SciML.md]:
```suggestion
forward/adjoint rules via the derivative rule package [ChainRules.jl](https://github.com/JuliaDiff/ChainRules.jl).

```
assuming you are specifically limiting this to ChainRules

File: _posts/2020-3-27-SciML.md [hunk 10]
```diff
@@ -0,0 +1,380 @@
...
+
+As demonstrated in the following animation, you can take our stochastic
+differential equation solvers and training a circuit to control the solution
```
Comment (nilshg) [inline _posts/2020-3-27-SciML.md]:
```suggestion
differential equation solvers and train a circuit to control the solution

```

File: _posts/2020-3-27-SciML.md [hunk 11]
```diff
@@ -0,0 +1,380 @@
...
+which is routinely tested with DifferentialEquations.jl and the machine learning
+AD tooling like Zygote.jl, meaning that you can be sure that our surrogates
+modeling tools can train on differential equations and then be used inside
+of deep learning stacks. It is this interconnectivity that will allow
+next-generation SciML methodologies to get productionized in a way that will
+impact "big science" and industrial use.
+
+### We keep up with advances in computational hardware to ensure compatibility with the latest high performance computing tools.
+
+Today, Intel CPUs and NVIDIA GPUs are the dominant platform, but that won't always
```
Comment (nilshg) [inline _posts/2020-3-27-SciML.md]:
```suggestion
Today, Intel CPUs and NVIDIA GPUs are the dominant platforms, but that won't always

```

File: _posts/2020-3-27-SciML.md [hunk 12]
```diff
@@ -0,0 +1,380 @@
...
+be the case. [One of the upcoming top supercomputers will be entirely AMD-based, with AMD CPUs and AMD GPUs](https://www.anandtech.com/show/15581/el-capitan-supercomputer-detailed-amd-cpus-gpus-2-exaflops).
+In addition, [Intel GPUs](https://www.anandtech.com/show/15120/intels-2021-exascale-vision-in-aurora-two-sapphire-rapids-cpus-with-six-ponte-vecchio-gpus)
+are scheduled to be a component in the following top supercomputers. We are
```
Comment (nilshg) [inline _posts/2020-3-27-SciML.md]:
```suggestion
are scheduled to be a component in future supercomputers. We are

```
(unless you actually wanted to list some of the "following" supercomputers?)

File: _posts/2020-3-27-SciML.md (hunk 7 above)
Comment (ChrisRackauckas) [reply _posts/2020-3-27-SciML.md]: It's somewhat a statement of fact, but really defining what we've had as a working principle in our head

File: _posts/2020-3-27-SciML.md [hunk 13]
```diff
@@ -0,0 +1,380 @@
...
+
+### We provide high-level domain-specific modeling tools to make scientific modeling more accessible
+
+A physics, biologist, chemist, etc. should be able to pick up our tools and make
+use of high performance scientific machine learning methods without requiring the
+understanding of every component. To make this a reality, we provide high-level
+domain-specific modeling tools as a frontend for building and generating models.
+[DiffEqBiological.jl](https://github.com/SciML/DiffEqBiological.jl) is a prime
+example which generates high performance simulations from a description of the
+chemical reactions. For example, the following solves the Michaelis-Menton model
```
[the remark this answers is not in the document]
Comment (ChrisRackauckas) [reply _posts/2020-3-27-SciML.md]: Exactly the response I want from this re-branding 👍 .

File: _posts/2020-3-27-SciML.md (hunk 9 above)
Comment (ChrisRackauckas) [reply _posts/2020-3-27-SciML.md]: we aren't specifically limiting it to ChainRules, but we hope everything coalesces

File: _posts/2020-3-27-SciML.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (ViralBShah) [reply _posts/2020-3-27-SciML.md]: Would you be renaming some of the packages too?

[the remark this answers is not in the document]
Comment (ChrisRackauckas) [reply _posts/2020-3-27-SciML.md]: I need to consider it. DiffEqBiological is a terrible name, DiffEqFlux is mostly not using Flux, and etc. Renaming packages is a little scary though, and I don't like to things that are semi-breaking to users unless they greatly benefit from it.

**Comment by ViralBShah** [comment]
Are DiffEqBiological and ModellingToolkit the only two DSLs in here? Seems like there is a general framework here: High level domain specific DSL -> IR -> diffeq sovlers. There's also some optimization/AD that flows through all of that.

Is that architecture worth discussing, maybe with a diagram?

**Comment by ChrisRackauckas** [comment]
There's also ParameterizedFunctions.jl and Pumas.jl. DiffEqBio has its own IR right now and it's one of my priorities to clean that up / fix it, so I'd rather not discuss all of the details on it quite yet. The full story of MTK isn't complete enough yet. I think that next year when we can say things like, take your bio model and here's the reduced approximation with parallelism, and do the same on Pumas, and the same in JuMP (/ whatever we call that MOI interface), all due to the same underlying mechanism, then that's a story that deserves to be discussed on its own.

**Comment by alanedelman** [comment]
may be a good reference for the first line

https://www.osti.gov/servlets/purl/1478744

File: _posts/2020-3-27-SciML.md [hunk 14]
```diff
@@ -0,0 +1,392 @@
...
+Scientific Machine Learning, abbreviated SciML, has been taking the academic
+world by storm as an interesting blend of traditional scientific mechanistic
+modeling (differential equations) with machine learning methodologies like
+deep learning. While traditional deep learning methodologies have had difficulties
+with interpretability and enforcing physical constraints, this blend with
+numerical analysis and differential equations has evolved into a field of research
+with new methods, architectures, and algorithms which overcome these traditional
+problems. Because of this, new software will be required in order to accommodate
+the connection between scientific simulators and scientific machine learning
+training loops.
```
Comment (MasonProtter) [inline _posts/2020-3-27-SciML.md]: Perhaps it's worth mentioning explicitly in the paragraph above that traditional deep learning approaches usually don't encode prior knowledge that the scientist might have about their system?

File: _posts/2020-3-27-SciML.md [hunk 15]
```diff
@@ -0,0 +1,392 @@
...
+
+SciML is an open source software organization for the development
+and maintenance of a feature-filled and high performance set of tooling for
+scientific machine learning. This includes the full gambit of tools from
```
Comment (MasonProtter) [inline _posts/2020-3-27-SciML.md]:
```suggestion
scientific machine learning. This includes the full gamut of tools from
```
I think gamut is more appropriate

File: _posts/2020-3-27-SciML.md [hunk 16]
```diff
@@ -0,0 +1,392 @@
...
+differential equation solvers to scientific simulators and tools for automatically
+discovering scientific models. What I wanted to do with this post was introduce
+
```
Comment (MasonProtter) [inline _posts/2020-3-27-SciML.md]:
```suggestion
```
don't think there was supposed to be a newline here?

File: _posts/2020-3-27-SciML.md [hunk 17]
```diff
@@ -0,0 +1,392 @@
...
+- Our next steps
+- How you can join in the process
+
+## The Software that SciML Provides
+
+### We provide best-in-class tooling for solving differential equations
+
+We will continue to have [DifferentialEquations.jl](https://docs.sciml.ai/dev/) at
+the core of the organization to support high performance solving of the differential
+
```
Comment (MasonProtter) [inline _posts/2020-3-27-SciML.md]:
```suggestion
```
Another unintentional newline?

File: _posts/2020-3-27-SciML.md [hunk 18]
```diff
@@ -0,0 +1,392 @@
...
+```
+
+Reorganizing under the SciML umbrella will make it easier for users to discover
+and apply our global sensitivity analysis methods outside of differential equation
+contexts, such as with neural networks.
+
+### We provide high-level domain-specific modeling tools to make scientific modeling more accessible
+
+A physicist, biologist, chemist, etc. should be able to pick up our tools and make
+
```
Comment (MasonProtter) [inline _posts/2020-3-27-SciML.md]:
```suggestion
```

File: _posts/2020-3-27-SciML.md [hunk 19]
```diff
@@ -0,0 +1,392 @@
...
+res1 = gsa(ishi,Sobol(),A,B)
+```
+
+Reorganizing under the SciML umbrella will make it easier for users to discover
+and apply our global sensitivity analysis methods outside of differential equation
+contexts, such as with neural networks.
+
+### We provide high-level domain-specific modeling tools to make scientific modeling more accessible
+
+A physicist, biologist, chemist, etc. should be able to pick up our tools and make
```
Comment (MasonProtter) [inline _posts/2020-3-27-SciML.md]:
```suggestion
Differential equations appear in nearly every scientific domain, but most
scientific domains have their own specialized idioms and terminology. 
A physicist, biologist, chemist, etc. should be able to pick up our tools and make
```

File: _posts/2020-3-27-SciML.md [hunk 20]
```diff
@@ -0,0 +1,392 @@
...
+
+use of high performance scientific machine learning methods without requiring the
+understanding of every component. To make this a reality, we provide high-level
```
Comment (MasonProtter) [inline _posts/2020-3-27-SciML.md]:
```suggestion
understanding of every component and using abstractions that make sense to
their field. To make this a reality, we provide high-level
```

File: _posts/2020-3-27-SciML.md [hunk 21]
```diff
@@ -0,0 +1,392 @@
...
+# Now let's draw a phase plot
+
+ut = numpy.transpose(sol.u)
+from mpl_toolkits.mplot3d import Axes3D
+fig = plt.figure()
+ax = fig.add_subplot(111, projection='3d')
+ax.plot(ut[0,:],ut[1,:],ut[2,:])
+plt.show()
+```
+
```
Comment (MasonProtter) [inline _posts/2020-3-27-SciML.md]: Maybe mention efforts / plans around static compilation in order to deliver julia interop to other languages with less friction.

File: _posts/2020-3-27-SciML.md [hunk 22]
```diff
@@ -0,0 +1,392 @@
...
+applications. Our issues and [chat channel](https://gitter.im/JuliaDiffEq/Lobby)
+serve as places to not just debug existing software, but also discuss new
+methods and help create high performance implementations.
+
+In addition, we support many student activities to bring new researchers into the
+community. Many of the maintainers of our packages, like Yingbo Ma, Vaibhav Dixit,
+Kanav Gupta, Kirill Zubov, etc. all started as one of our over 50 student developers
+from past [Google Summer of Code](https://summerofcode.withgoogle.com/) and other
+[Julia Seasons of Contributions](https://julialang.org/jsoc/).
+
```
Comment (MasonProtter) [inline _posts/2020-3-27-SciML.md]: Maybe mention that the fact that everything is written in julia makes it easier for students to 'look under the hood' and start understanding and tweaking libraries without learning 5 different languages.

File: _posts/2020-3-27-SciML.md [hunk 23]
```diff
@@ -0,0 +1,392 @@
...
+
+As demonstrated in the following animation, you can take our stochastic
+differential equation solvers and train a circuit to control the solution
+
+by simply piecing together compatible packages.
+
+![](https://user-images.githubusercontent.com/1814174/51399524-2c6abf80-1b14-11e9-96ae-0192f7debd03.gif)
+
+### Performance is considered a priority, and performance issues are considered a bug
```
Comment (MasonProtter) [inline _posts/2020-3-27-SciML.md]:
```suggestion
### Performance is considered a priority, and performance issues are considered bugs
```

File: _posts/2020-3-27-SciML.md [hunk 24]
```diff
@@ -0,0 +1,392 @@
...
+AD tooling like Zygote.jl, meaning that you can be sure that our surrogates
+modeling tools can train on differential equations and then be used inside
+of deep learning stacks. It is this interconnectivity that will allow
+next-generation SciML methodologies to get productionized in a way that will
+impact "big science" and industrial use.
+
+### We keep up with advances in computational hardware to ensure compatibility with the latest high performance computing tools.
+
+Today, Intel CPUs and NVIDIA GPUs are the dominant platforms, but that won't always
+
```
Comment (MasonProtter) [inline _posts/2020-3-27-SciML.md]:
```suggestion
```

File: _posts/2020-3-27-SciML.md [hunk 25]
```diff
@@ -0,0 +1,392 @@
...
+be the case. [One of the upcoming top supercomputers will be entirely AMD-based, with AMD CPUs and AMD GPUs](https://www.anandtech.com/show/15581/el-capitan-supercomputer-detailed-amd-cpus-gpus-2-exaflops).
+In addition, [Intel GPUs](https://www.anandtech.com/show/15120/intels-2021-exascale-vision-in-aurora-two-sapphire-rapids-cpus-with-six-ponte-vecchio-gpus)
+are scheduled to be a component in future supercomputers. We are
+
```
Comment (MasonProtter) [inline _posts/2020-3-27-SciML.md]:
```suggestion
```

File: _posts/2020-3-27-SciML.md [hunk 26]
```diff
@@ -0,0 +1,392 @@
...
+committed to maintaining a SciML toolchain that works on all major platforms,
+updating our compiler backends as new technology is released.
```
Comment (MasonProtter) [inline _posts/2020-3-27-SciML.md]: Maybe link to https://github.com/jpsamaroo/ROCArrays.jl as a potential avenue.

File: _posts/2020-3-27-SciML.md [hunk 27]
```diff
@@ -0,0 +1,415 @@
...
+layout: post
+title:  "SciML: An Open Source Software Organization for Scientific Machine Learning"
+date:   2020-3-27 10:00:00
+categories:
+---
+
+Computational scientific discovery is at an interesting juncture. While we have
+mechanistic models of lots of different scientific phenomena, and reams of data
+being generated from experiments - our computational capabilities are unable to
+keep up. Our problems are too large for realistic simulation. Our problems run
```
Comment (MasonProtter) [inline _posts/2020-3-27-SciML.md]:
```suggestion
keep up. Our problems are too large for realistic simulation. Our problems
```

File: _posts/2020-3-27-SciML.md [hunk 28]
```diff
@@ -8,26 +8,38 @@ categories:
...
 problems while adding the data-driven automatic learning features of modern
-deep learning. However, new software will be required in order to accommodate
-the connection between scientific simulators and scientific machine learning
-training loops.
+deep learning. Many successes have already been found, with tools like
+[physics-informed neural networks](https://www.sciencedirect.com/science/article/pii/S0021999118307125),
+[deep BSDE solvers for high dimensional PDEs](https://link.springer.com/article/10.1007%2Fs40304-017-0117-6),
+and [neural surrogates](https://arxiv.org/pdf/2001.08055.pdf) showcasing how
+deep learning can greatly improve scientific modeling practice. At the same time,
+researcher are quickly finding that our training techniques will need to be
```
Comment (ViralBShah) [inline _posts/2020-3-27-SciML.md]: researchers

File: _posts/2020-3-27-SciML.md [hunk 29]
```diff
@@ -0,0 +1,415 @@
...
+consider that an issue and should get it fixed. High performance is required for
+scientific machine learning to scale, and so we take performance seriously.
+
+### Our packages are routinely and robustly tested with the tools for both scientific simulation and machine learning
+
+This means we will continue to develop tools like
+[DiffEqFlux.jl](https://github.com/SciML/DiffEqFlux.jl) which supports the
+connection between the [DifferentialEquations.jl](https://docs.sciml.ai/dev/)
+differential equation solvers and the [Flux](https://fluxml.ai/) deep learning
+library. Other examples this includes our
```
Comment (felixcremer) [inline _posts/2020-3-27-SciML.md]: Should this be:Other examples include ?

**Pull request merged**